awx/main/tasks/callback.py

```diff
@@ -0,0 +1,203 @@
+import json
+import time
+import logging
+from collections import deque
+
+# Django
+from django.utils.timezone import now
+from django.conf import settings
+from django_guid.middleware import GuidMiddleware
+
+
+# AWX
+from awx.main.tasks.jobs import RunInventoryUpdate, RunProjectUpdate
+from awx.main.redact import UriCleaner
+from awx.main.constants import (
+    MINIMAL_EVENTS,
+)
+
+
+logger = logging.getLogger('awx.main.tasks.jobs')
+
+
+class RunnerCallback:
+    def __init__(self):
+        self.runner_callback.parent_workflow_job_id = None
+        self.runner_callback.host_map = {}
+        self.runner_callback.guid = GuidMiddleware.get_guid()
+        self.runner_callback.job_created = None
+        self.runner_callback.recent_event_timings = deque(maxlen=settings.MAX_WEBSOCKET_EVENT_RATE)
```

Comment:
Pattern-wise, `self.runner_callback.parent_workflow_job_id = None` --> `self.parent_workflow_job_id = None`

This is possibly the most important thing to clear up so that you can get running with testing this.

Reply:
I have made the changes.